content/en/docs/guides/performance.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Performance Best Practices
+description: A user guide of both general and language-specific best practices to improve performance.
+---
+
+### General
+
+*   Always **re-use stubs and channels** when possible.
+
+*   **Use keepalive pings** to keep HTTP/2 connections alive during periods of
+    inactivity to allow initial RPCs to be made quickly without a delay (i.e.
+    C++ channel arg GRPC_ARG_KEEPALIVE_TIME_MS). 
+
+*   **Use streaming RPCs** when wanting to avoid continuous RPC initiation,
+    which includes connection load balancing at the client-side, starting a new
+    HTTP/2 request at the transport layer, and invoking a user-defined method
+    handler on the server side. Streams are particularly well-suited to handling
+    a long-lived logical flow of data from the client-to-server,
+    server-to-client, or in both directions. Streams, however, cannot be load
+    balanced once they have started and can be hard to debug for stream
+    failures. \
+    ***Side note:*** *This does not apply to Python (see Python section for
+    details).*
+
+*   Each gRPC channel uses 0 or more HTTP/2 connections and each connection
+    usually has a limit on the number of concurrent streams. When the number of
+    active RPCs on the connection reaches this limit, additional RPCs are queued
+    in the client and must wait for active RPCs to finish before they are sent.
+    Applications with high load or long-lived streaming RPCs might see
+    performance issues because of this queueing. There are two possible
+    solutions:
+
+    1.  **Create a separate channel for each area of high load** in the
+        application, or
+
+    2.  **Use a pool of gRPC channels** to randomly distribute RPCs over
```

Comment:
s/randomly//. Round robin is common and isn't random. I know that isn't what this is actually meaning, but using random here may be misleading and doesn't seem to add value.